tools/check-style.sh

```diff
@@ -11,72 +11,60 @@
 # 6. opening brace on its own line. It should always be on the same line as the
 #    if/while/for/do statment.
-# 
+#
 # Invoke as: tools/check-style.sh
 #
 
-errors=0
+check_style_errors=0
 IFS=$'\n'
-found=
-# The mt=41 sets a red background for matched tabs:
-GREP_COLORS='mt=41' GREP_COLOR='41' grep $'\t' include tests/*.{cpp,py,h} docs/*.rst -rn --color=always |
-        while read f; do
-    if [ -z "$found" ]; then
-        echo -e '\033[31m\033[01mError: found tabs instead of spaces in the following files:\033[0m'
-        found=1
-        errors=1
-    fi
 
-    echo "    $f"
-done
-
-found=
-grep -IUlr $'\r' include tests/*.{cpp,py,h} docs/*.rst --color=always |
-        while read f; do
-    if [ -z "$found" ]; then
-        echo -e '\033[31m\033[01mError: found CRLF characters in the following files:\033[0m'
-        found=1
-        errors=1
-    fi
-
-    echo "    $f"
-done
+found="$( GREP_COLORS='mt=41' GREP_COLOR='41' grep $'\t' include tests/*.{cpp,py,h} docs/*.rst -rn --color=always )"
+if [ ! -z "$found" ]; then
+    # The mt=41 sets a red background for matched tabs:
+    echo -e '\033[31m\033[01mError: found CRLF characters in the following files:\033[0m'
+    check_style_errors=1
+    echo "$found" | while read line; do
+        echo -e '\033[31m\033[01mError: found tabs instead of spaces in the following files:\033[0m'
+    done
+fi
 
-found=
-# The mt=41 sets a red background for matched trailing spaces
-GREP_COLORS='mt=41' GREP_COLOR='41' grep '[[:blank:]]\+$' include tests/*.{cpp,py,h} docs/*.rst -rn --color=always |
-        while read f; do
-    if [ -z "$found" ]; then
-        echo -e '\033[31m\033[01mError: found trailing spaces in the following files:\033[0m'
-        found=1
-        errors=1
-    fi
 
-    echo "    $f"
-done
+found="$( grep -IUlr $'\r' include tests/*.{cpp,py,h} docs/*.rst --color=always )"
+if [ ! -z "$found" ]; then
+    echo -e '\033[31m\033[01mError: found CRLF characters in the following files:\033[0m'
+    check_style_errors=1
+    echo "$found" | while read line; do
+        echo "    $line"
+    done
+fi
 
-found=
-grep '\<\(if\|for\|while\|catch\)(\|){' include tests/*.{cpp,py,h} -rn --color=always |
-        while read line; do
-    if [ -z "$found" ]; then
-        echo -e '\033[31m\033[01mError: found the following coding style problems:\033[0m'
-        found=1
-        errors=1
-    fi
+found="$(GREP_COLORS='mt=41' GREP_COLOR='41' grep '[[:blank:]]\+$' include tests/*.{cpp,py,h} docs/*.rst -rn --color=always )"
+if [ ! -z "$found" ]; then
+    # The mt=41 sets a red background for matched trailing spaces
+    echo -e '\033[31m\033[01mError: found trailing spaces in the following files:\033[0m'
+    check_style_errors=1
+    echo "$found" | while read line; do
+        echo "    $line"
+    done
+fi
 
-    echo "    $line"
-done
+found="$(grep '\<\(if\|for\|while\|catch\)(\|){' include tests/*.{cpp,py,h} -rn --color=always)"
```

Comment:
I think it's a mistake (in the existing code, copied over here) to run this test again `tests/*.py` as far as I can see; let's remove the `,py` from the grep argument input files.

Reply:
Done.